framework/src/meshgenerators/FlipSidesetGenerator.C

```diff
@@ -0,0 +1,72 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FlipSidesetGenerator.h"
+
+#include "CastUniquePointer.h"
+
+registerMooseObject("MooseApp", FlipSidesetGenerator);
+
+InputParameters
+FlipSidesetGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+  params.addClassDescription("A Mesh Generator which flips a given sideset");
+  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
+  params.addRequiredParam<BoundaryName>("boundary", "The sideset (boundary) that will be flipped");
+  return params;
+}
+
+FlipSidesetGenerator::FlipSidesetGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _input(getMesh("input")),
+    _sideset_name(getParam<BoundaryName>("boundary"))
+{
+}
+
+std::unique_ptr<MeshBase>
+FlipSidesetGenerator::generate()
+{
+  // get boundary info
+  BoundaryInfo & boundary_info = _input->get_boundary_info();
+  // get id of the input sideset
+  boundary_id_type sideset_id = boundary_info.get_id_by_name(_sideset_name);
+
+  // Throw an error if the sideset doesn't exist
+  if (sideset_id == libMesh::BoundaryInfo::invalid_id)
+    mooseError("sideset doesn't exist in mesh");
```

Comment:
It would be good to put the requested name here, for users who use these multiple times in one input.